numba/core/dispatcher.py

```diff
@@ -23,6 +25,31 @@
 import numba.core.event as ev
 
 
+class TargetConfig:
+    _tls = threading.local()
+
+    def __init__(self):
+        tls = self._tls
+        try:
+            tls_stack = tls.stack
+        except AttributeError:
+            tls_stack = tls.stack = list()
+
+        self._stack = tls_stack
+
+    def push(self, state):
+        self._stack.append(state)
+
+    def pop(self):
+        return self._stack.pop()
+
+    def get(self):
+        return self._stack[0]
```

Comment:
push/pop works with the tail but get works with head. Is it correct?

Reply:
that's a mistake. it should be getting from the tail.